midterm-2023-11-06/code/5088-darko-petrevski/script.js

```diff
@@ -0,0 +1,126 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const bandsTable = document.getElementById('bandsTable');
+    const nameFilter = document.getElementById('nameFilter');
+    const countryFilter = document.getElementById('countryFilter');
+    const genreFilter = document.getElementById('genreFilter');
+    const yearActiveFilter = document.getElementById('yearActiveFilter');
+
+    let bandsData = []; 
+
+    fetch('https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json')
+        .then(response => response.json())
+        .then(data => {
+            bandsData = data.metalBands;
+            populateTable(bandsData);
+            populateFilterDropdowns(bandsData);
+        })
+        .catch(error => console.error('Error fetching data:', error));
+
+    function populateTable(data) {
+        const table = document.getElementById('bandsTable');
+    
+        table.innerHTML = '';
+    
+        const headers = [
+            'ID',
+            'Name',
+            'Formed',
+            'Location',
+            'Country',
+            'Genre',
+            'Members',
+            'Albums',
+            'First Album',
+            'Last Album',
+            'Years Active'
+        ];
+    
+        const headerRow = table.insertRow();
+        headers.forEach(headerText => {
+            const th = document.createElement('th');
+            th.textContent = headerText;
+            headerRow.appendChild(th);
+        });
+    
+        data.forEach(band => {
+            const row = table.insertRow();
+            row.innerHTML = `
+                <td>${band.id}</td>
+                <td>${band.name}</td>
+                <td>${band.formed}</td>
+                <td>${band.location}</td>
+                <td>${band.location.split(', ')[1]}</td>
+                <td>${band.genre}</td>
+                <td>${formatMembers(band.members)}</td>
+                <td>${band.albums.length}</td>
+                <td>${band.albums[0].name} (${band.albums[0].year})</td>
+                <td>${band.albums[band.albums.length - 1].name} (${band.albums[band.albums.length - 1].year})</td>
+                <td>${getYearsActive(band)}</td>
+            `;
+        });
+    }
+    
+    function formatMembers(members) {
+        if (members.length > 5) {
+            members = members.slice(0, 5).sort();
+            members[members.length - 1] = members[members.length - 1].replace(/,/, ' &');
```

Comment:
Not sure that this will work, the `,` and `&` characters are not part of the band members names.